### PR TITLE
Create Models for Project

### PR DESCRIPTION
**Pull request by RaoAkif**
### Add Migrations and Create Models for Project
- Create Models for User, Car, and Reservation
- Create Associations (User has many Cars, Car belongs to Reservations, User has many Reservations)
- Add Validations to the Models

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  has_many :reservations
+  has_many :cars, through: :reservations
+
```
Comment (Retky) [inline app/models/user.rb:4]: 3| Great implementation to facilitate selectors. 🔥

File: db/migrate/20220811152628_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+      t.string :name
+
```
Comment (Retky) [inline db/migrate/20220811152628_create_users.rb]: [Optional] You could make the `name` required for the db too by:
3|       `t.string :name` -> `t.string :name, null: false`

**Review by Retky — APPROVED** [review]
## Hello @RaoAkif 👍🏻,
Excellent work! 💯
You have done a great job. 🥳
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit:
Congratulations!!!🎊

### HIGHLIGHTS 🌟 
- ✔️ Use of best practices
- ✔️ Clean code
- ✔️ Good code organization
 
# Status: Approved ✔️ 

oscars-standing-ovation


### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
---

File: db/migrate/20220811152628_create_users.rb (hunk 2 above)
Comment (RaoAkif) [reply db/migrate/20220811152628_create_users.rb]: Updated `t.string :name` -> `t.string :name, null: false`

**Pull request merged**